spec/ameba/rule/lint/debugger_statement_spec.cr

```diff
@@ -18,16 +18,18 @@ module Ameba::Rule::Lint
           end
         end
         A.new.debugger
-      )
+        CRYSTAL
     end
 
     it "fails if there is a debugger statement" do
-      expect_issue subject, %(
-        a = 2
-        debugger
-        # ^{} error: Possible forgotten debugger statement detected
-        a = a + 1
-      )
+      source = expect_issue subject, <<-CRYSTAL
+          a = 2
+          debugger
+        # ^^^^^^^^ error: Possible forgotten debugger statement detected
+          a = a + 1
```

Comment:
Indenting all of the code to accommodate the difference in comment seems pretty off.
I like the old way with `# ^{}` construct more, to be honest.

Reply:
Thanks, I wasn't sure about making this change. RuboCop reserves `^{}` for zero-length cop offenses, but in Ameba it's actually impossible for issues to have zero length since we always add `+ 1` to the end position. So I think it is acceptable to deviate from RuboCop and have `# ^{}` mean "this issue starts at column 1 or 2". However, I'll throw out some ideas because why not?

Option 1:

```
debugger
# ^{} error: Possible forgotten debugger statement detected
```

Option 2: More carets

```
debugger
# ^{}^^^ error: Possible forgotten debugger statement detected
```

Option 3: Replace space after `#` with caret

```
debugger
#^{}^^^^ error: Possible forgotten debugger statement detected
```

Option 4: All carets

```
debugger
#^^^^^^^ error: Possible forgotten debugger statement detected
```

Reply:
Thanks for providing a rationale for this change and the comparison with RuboCop! ❤️ 
I like the alternatives, yet I'm missing my favorite, so I'll throw one more option up for discussion :)

Option 5: All carets with space

```cr
debugger
# ^^^^^^ error: Possible forgotten debugger statement detected
```

Reply:
There’s also another option that I’ve been reluctant to consider:

Option 6: Last Resort

```cr
debugger # error: Possible forgotten debugger statement detected
```

This is already possible in `master`, we just haven’t used it yet. I’ve been avoiding using it because it has no precision other than the line number that it’s on. However, given all the other options we’ve considered, this is probably the simplest and/or cleanest one. I’m not sure though, so to avoid bikeshedding, I’ll defer to you or @veelenga.

Reply:
Yeah, Option 6 seems to be the least accurate, that's no good.

My picks are (ordered by the preference factor):
- Option 5
- Option 2
- Option 1

Reply:
I agree with @Sija. Except "Option 2" because it looks unnecessary to add more carets if there is already that indicator of full-line issue `^{}`

Reply:
Shall we go forward with no. 5 then?

Reply:
I'm good for 5 or 1. If you prefer 5, let's do it. Thanks guys 👏

Reply:
I've pushed a commit that does both: carets are added if the issue is long enough or `^{}` if not.

Old:

```cr
  debugger
# ^^^^^^^^ error: Possible forgotten debugger statement detected
  require "big"
# ^ error: Duplicated require of `big`
```

New:

```cr
debugger
# ^^^^^^ error: Possible forgotten debugger statement detected
require "big"
# ^{} error: Duplicated require of `big`
```